restcomm/restcomm.ussd/src/main/java/org/restcomm/connect/ussd/interpreter/UssdInterpreter.java

```diff
@@ -545,27 +545,34 @@ else if ("outbound-api".equals(response.get().direction())){
             }
         } else if (DownloaderResponse.class.equals(klass)) {
             final DownloaderResponse response = (DownloaderResponse) message;
-
-            int sc = response.get().getStatusCode();
-            //processingInfoRequest
             if (logger.isDebugEnabled()) {
-                logger.debug("Rcml URI : " + response.get().getURI() + "response success=" + response.succeeded()
-                    + ", statusCode=" + response.get().getStatusCode()+" state="+state);
+                logger.debug("Rcml DownloaderResponse success=" + response.succeeded());
             }
             //FIXME: what if these ifblocks arent in downloadingRcml?
             if (response.succeeded()) {
+                int sc = response.get().getStatusCode();
+                //processingInfoRequest
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Rcml DownloaderResponse URI : " + response.get().getURI()
+                        + ", statusCode=" + response.get().getStatusCode()+" state="+state);
+                }
                 if(HttpStatus.SC_OK == sc){
                     fsm.transition(message, ready);
                 } else if (HttpStatus.SC_NOT_FOUND == sc){
                     fsm.transition(message, notFound);
                 }
             } else {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Rcml DownloaderResponse error : " + response.error()
+                        + ", cause=" + response.cause().getMessage());
+                }
                 if (downloadingRcml.equals(state) && fallbackUrl!=null) {
                     fsm.transition(message, downloadingFallbackRcml);
                 } else {
+                    //TODO: should we rethrow response.error()?
```

Comment:
exception handling is not always rethrowing... In this case we are both sending BYE potentially,and letting the FSM transition, so i would say no rethrow in this case...

Reply:
should we remove the TODO?

Reply:
i would certainly remvoe the TODO here

Reply:
DONE